src/main/java/edu/umich/its/cpm/MigrationController.java

```diff
@@ -258,20 +258,47 @@ public Migration migrated(HttpServletRequest request) {
 	 * 
 	 * @return
 	 */
-	private String DEVELOPER_TOKEN = "";
-	private static final int MAX_DEPTH = 1;
-
 	@RequestMapping("/box/folders")
-	public String getBoxFolders() {
-
+	public void getBoxFolders(HttpServletRequest request, HttpServletResponse response) {
+		
+		String boxClientId = env.getProperty("box_client_id");
+		String boxClientSecret = env.getProperty("box_client_secret");
+		String boxClientRedirectUri = env.getProperty("box_client_redirect_uri") + "/box/folders_authorized";
+		
+		// need to have all Box app configurations
+		if (boxClientId == null || boxClientSecret == null || boxClientRedirectUri == null)
+		{
+			log.error("Missing box integration parameters");
+			return;
```

Comment:
If this is REST the status code needs to be set to an appropriate value. 500 Internal Server Error sounds right for this one.